pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -197,6 +197,7 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
       String tableName = entry.getKey();
       Set<String> unavailableSegments = entry.getValue();
       numUnavailableSegments += unavailableSegments.size();
+      brokerResponse.setPartialResult(true);
       brokerResponse.addToExceptions(new QueryProcessingException(QueryException.SERVER_SEGMENT_MISSING_ERROR_CODE,
```

Comment:
If some servers are not available we currently used to throw an Exception instead of the results
So likely this won't be required as we have an exception stating which servers are in error